api/pkg/backend/service.go

```diff
@@ -230,9 +258,15 @@ func (s *APIService) DeleteBackend(request *restful.Request, response *restful.R
 	}
 	id := request.PathParameter("id")
 	log.Logf("Received request for deleting backend: %s\n", id)
-	ctx := context.Background()
-	owner := "test"
-	res, err := s.s3Client.ListBuckets(ctx, &s3.BaseRequest{Id: owner})
+
+	actx := request.Attribute(c.KContext).(*c.Context)
```

Comment:
there are few set of lines which are repeating in the code, we can write a function or a module to call these everytime, instead of writing them repeatedly

Reply:
Can we optimize it?

Reply:
Do you mean 'ctx := metadata.NewContext(....)', it's actually one statement, but it's a little long, so it's breaked to several lines.

Reply:
BTW, different APIs may need to add different context.

Reply:
fixed